src/metabase/api/card.clj

```diff
@@ -1128,6 +1133,19 @@ saved later when it is ready."
    query     ms/NonBlankString}
   (param-values (api/read-check Card card-id) param-key query))
 
+(defn- scan-and-sync-table!
+  [{:keys [is_on_demand is_full_sync] :as database} table]
+  (sync-fields/sync-fields-for-table! database table)
+  (when (or is_on_demand is_full_sync)
+    (future (sync/sync-table! table))))
+
+(defn- syncable-schema?
+  [driver database schema-name]
+  (or (nil? schema-name)
+      (let [schema-filter-prop                      (driver.u/find-schema-filters-prop driver)
+            [inclusion-patterns exclusion-patterns] (driver.s/db-details->schema-filter-patterns (:name schema-filter-prop) database)]
```

Comment:
I've seen this exact sequence of 2 lines a few times, and written them myself in tests. Wonder if it's time to extract a function and put it in the `driver.s` namespace for reuse. Also the function can just take a database instance not a driver, since the driver is on the `:engine` key.

Reply:
did some tidying here https://github.com/metabase/metabase/pull/31194/commits/3c8c6d2e8dd51fcaa5d01a655f82f81a8a4661bf